Bottom: 01e3c2c0f37572e9555ececfa38d5f92a80627fb
Top:    7789841c5305f1e6ee8dd914ee3e43bfffbb7ba3
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-05-21 17:52:20 -0700

Refresh of rs-opt-qpsize

---

diff --git a/src/rsocket.c b/src/rsocket.c
index 9d373b9..893cc57 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -1695,6 +1695,20 @@ int rsetsockopt(int socket, int level, int optname,
 		}
 		break;
 	case SOL_RDMA:
+		if (rs->state != rs_init && rs->state != rs_bound) {
+
+		switch (optname) {
+		case RDMA_SQSIZE:
+			*((int *) optval) = rs->sq_size;
+			*optlen = sizeof(int);
+			break;
+		case RDMA_RQSIZE:
+			*((int *) optval) = rs->rq_size;
+			*optlen = sizeof(int);
+			break;
+		default:
+			break;
+		}
 		break;
 	default:
 		break;
@@ -1762,7 +1776,19 @@ int rgetsockopt(int socket, int level, int optname,
 		}
 		break;
 	case SOL_RDMA:
-		ret = ENOTSUP;
+		switch (optname) {
+		case RDMA_SQSIZE:
+			*((int *) optval) = rs->sq_size;
+			*optlen = sizeof(int);
+			break;
+		case RDMA_RQSIZE:
+			*((int *) optval) = rs->rq_size;
+			*optlen = sizeof(int);
+			break;
+		default:
+			ret = ENOTSUP;
+			break;
+		}
 		break;
 	default:
 		ret = ENOTSUP;
